### PR TITLE
Freeze 3.73 schema in migrator

### PR DESCRIPTION
**Pull request by c-du**
## Description

This PR wont merge before 3.73 branching out.
Freeze 3.73 schema in migrator.
I did not remove the auto generator because I want to keep that in a clean PR dedicated for stopping generators so we can conveniently re-use that PR in case we want to make massive change.

## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: tools/generate-helpers/pg-table-bindings/migration.go.tpl [hunk 1]
```diff
@@ -19,7 +19,8 @@ import (
 	pgStore "github.com/stackrox/rox/migrator/migrations/{{.Migration.Dir}}/postgres"
 	"github.com/stackrox/rox/migrator/types"
 	pkgMigrations "github.com/stackrox/rox/pkg/migrations"
-	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	pkgSchema "github.com/stackrox/rox/migrator/migrations/frozenschema/v73"
```
Comment (dashrews78) [inline tools/generate-helpers/pg-table-bindings/migration.go.tpl:22]: I think we should call this frozenSchema to make it stand out.

**Comment by dashrews78** [comment]
a README or something might be useful to explain the reasoning for separating the schema as well as how migrations work going forward.  Things like when we lock the migration schema, how we work migrations for the next release.  How that impacts the generated code.

**Comment by c-du** [comment]
> a README or something might be useful to explain the reasoning for separating the schema as well as how migrations work going forward. Things like when we lock the migration schema, how we work migrations for the next release. How that impacts the generated code.

Nice idea. Will do

**Review by dashrews78 — APPROVED** [review]
LGTM

File: migrator/migrations/frozenschema/v73/groups.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	CreateTableGroupsStmt = &postgres.CreateStmts{
+		GormModel: (*Groups)(nil),
+		Children:  []*postgres.CreateStmts{},
+	}
+
+	// GroupsSchema is the go schema for table `groups`.
+	GroupsSchema = func() *walker.Schema {
+		schema := walker.Walk(reflect.TypeOf((*storage.Group)(nil)), "groups")
+		return schema
+	}()
```
Comment (md2119) [inline migrator/migrations/frozenschema/v73/groups.go:24]: The schema seems different. The registration to global table is missing:
```	
GroupsSchema = func() *walker.Schema {
		schema := GetSchemaForTable("groups")
		if schema != nil {
			return schema
		}
		schema = walker.Walk(reflect.TypeOf((*storage.Group)(nil)), "groups")
		RegisterTable(schema, CreateTableGroupsStmt)
		return schema
	}()

```

**Review by md2119 — CHANGES_REQUESTED** [review]
The frozen schema seems to be differ than the ones in `pkg/postgres/schema`.

File: migrator/migrations/frozenschema/v73/groups.go (hunk 2 above)
Comment (c-du) [reply migrator/migrations/frozenschema/v73/groups.go:24]: That is on purpose. The dependency is pre-defined and the tables are not to be reused.

Comment (md2119) [reply migrator/migrations/frozenschema/v73/groups.go:24]: Okay. What I am worried about is that event if the migrator and its schema is frozen, one can still make changes to the schema/proto which actually will go unhandled. I think we should add a check which scan `pkg/schema/` dir for any changes. If the changes are desirable, the file is added to a whitelist. WDYT?

Comment (c-du) [reply migrator/migrations/frozenschema/v73/groups.go:24]: The migration should just have the schema at 3.73. It should not contain future changes to schema. That is what we expect.

**Review by md2119 — APPROVED** [review]
Approving because I do not want to block the work, but I feel a required check that fails the CI would be better for maintenance than dropping schema into new pkg.

**Pull request merged**